wagtail/wagtailcore/blocks/list_block.py

```diff
@@ -2,23 +2,39 @@
 
 from django import forms
 from django.contrib.staticfiles.templatetags.staticfiles import static
-from django.core.exceptions import ValidationError
+from django.core.exceptions import NON_FIELD_ERRORS, ValidationError
 from django.forms.utils import ErrorList
 from django.template.loader import render_to_string
 from django.utils.html import format_html, format_html_join
 from django.utils.safestring import mark_safe
+from django.utils.translation import ugettext as _
 
 from wagtail.wagtailcore.utils import escape_script
 
 from .base import Block
 from .utils import js_dict
 
-__all__ = ['ListBlock']
+__all__ = ['ListBlock', 'ListBlockValidationError']
+
+
+class ListBlockValidationError(ValidationError):
+    def __init__(self, block_errors=None, non_block_errors=None):
+        params = {}
+        if block_errors:
+            params.update(block_errors)
+        if non_block_errors:
+            params[NON_FIELD_ERRORS] = non_block_errors
+        super(ListBlockValidationError, self).__init__(
+            'Validation error in ListBlock', params=params)
```

Comment:
Suppose this can be better yes, as well for `StreamBlockValidationError`. The only goal of this custom ValidationError is so separate the block from the non-blocks errors. Although I see a lot in Django `ValidationError` the `params` argument is used a lot.